bob-backend/src/pearl/holder.rs

```diff
@@ -213,11 +213,11 @@ impl Holder {
 
     // @TODO remove redundant return result
     #[allow(clippy::cast_possible_truncation)]
-    async fn write_disk(storage: PearlStorage, key: Key, data: BobData) -> BackendResult<()> {
+    async fn write_disk(storage: PearlStorage, key: Key, data: &BobData) -> BackendResult<()> {
         counter!(PEARL_PUT_COUNTER, 1);
         let data_size = Self::calc_data_size(&data);
         let timer = Instant::now();
-        let res = storage.write(key, Data::from(data).to_vec()).await;
+        let res = storage.write(key, data.inner().to_vec()).await;
```

Comment:
It's unfortunate, but `Data` did the concatenation of the timestamp and the actual data. This was a bad design, but we cannot change it in the current release, because that will brake BLOB format compatibility. I suggest to implement the same behaviour on `BobData` with a separate function